exporter/exporterqueue/queue.go

```diff
@@ -70,14 +70,14 @@ type PersistentQueueSettings[T any] struct {
 // If cfg.StorageID is nil then it falls back to memory queue.
 // Experimental: This API is at the early stage of development and may change without backward compatibility
 // until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
-func NewPersistentQueueFactory[T itemsCounter](storageID *component.ID, factorySettings PersistentQueueSettings[T]) Factory[T] {
+func NewPersistentQueueFactory[T any](storageID *component.ID, factorySettings PersistentQueueSettings[T]) Factory[T] {
```

Comment:
We still want to introduce other queue measurements going forward, right? So this would need to be brought back to an interface implementing `ItemsCount() int` which will be a breaking change technically...

Reply:
We will accept a `Sizer` instead correct? We will pass the Request, so no need to enforce the request implements this API, the implementation can do what they want.

Reply:
Ok. Sounds good